docarray/array/storage/elastic/backend.py

```diff
@@ -0,0 +1,217 @@
+import copy
+import itertools
+import warnings
+from dataclasses import dataclass, field
+from typing import (
+    Dict,
+    Optional,
+    TYPE_CHECKING,
+    Union,
+    Tuple,
+    List,
+    Sequence,
+    Generator,
+    Iterator,
+)
+
+import numpy as np
+import uuid
+
+from ..base.backend import BaseBackendMixin
+from .... import DocumentArray, Document
+from ....helper import dataclass_from_dict
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+    from ....types import DocumentArraySourceType, ArrayType
+
+from docarray.math.helper import EPSILON
+
+
+def _sanitize_table_name(table_name: str) -> str:
+    ret = ''.join(c for c in table_name if c.isalnum() or c == '_')
+    if ret != table_name:
+        warnings.warn(f'The table name is changed to {ret} due to illegal characters')
+    return ret
+
+
+@dataclass
+class ElasticConfig:
+    n_dim: int  # dims  in elastic
+    basic_auth: Optional[Tuple[str, str]] = None
+    ca_certs: Optional[str] = None
+    distance: str = 'l2_norm'  # similarity in elastic
```

Comment:
what about using `cosine` as the default distance metric

Reply:
makes sense, done it